packages/web-forms/src/components/controls/RankControl.vue

```diff
@@ -0,0 +1,209 @@
+	highlight.index.value = index;
+
+	if (highlight.timeoutID) {
+		clearTimeout(highlight.timeoutID);
+		highlight.timeoutID = null;
+	}
+
+	if (highlight.index.value !== null) {
+		highlight.timeoutID = setTimeout(() => setHighlight(null), 1000);
+	}
+};
+
+const moveUp = (index: number) => {
+	const newPosition = index - 1;
+	if (newPosition < 0) {
+		return;
+	}
+	swapItems(index, newPosition);
+};
+
+const moveDown = (index: number) => {
+	const newPosition = index + 1;
+	if (newPosition >= options.value.length) {
+		return;
+	}
+	swapItems(index, newPosition);
+};
+
+const swapItems = (index: number, newPosition: number) => {
+	setHighlight(index);
+	const temp = options.value[index];
+	options.value[index] = options.value[newPosition];
+	options.value[newPosition] = temp;
+	setHighlight(newPosition);
+};
+</script>
+
+<template>
+	<ControlText :question="question" />
+
+	<VueDraggable
+		:id="question.nodeId"
+		v-model="options"
+		:delay="HOLD_DELAY"
+		:disabled="question.currentState.readonly"
+		ghost-class="fade-moving"
+		class="rank-control"
+		@update="setValues"
+	>
+		<div
+			v-for="(option, index) in options"
+			:id="option.value"
+			:key="option.value"
+			class="rank-option"
+			:class="{ 'moving': highlight.index.value === index }"
+			tabindex="0"
+			@keydown.up.prevent="moveUp(index)"
+			@keydown.down.prevent="moveDown(index)"
+		>
+			<div class="rank-label">
+				<svg xmlns="http://www.w3.org/2000/svg" width="25" height="25" viewBox="0 0 768 768">
+					<path d="M480 511.5q25.5 0 45 19.5t19.5 45-19.5 45-45 19.5-45-19.5-19.5-45 19.5-45 45-19.5zM480 319.5q25.5 0 45 19.5t19.5 45-19.5 45-45 19.5-45-19.5-19.5-45 19.5-45 45-19.5zM480 256.5q-25.5 0-45-19.5t-19.5-45 19.5-45 45-19.5 45 19.5 19.5 45-19.5 45-45 19.5zM288 127.5q25.5 0 45 19.5t19.5 45-19.5 45-45 19.5-45-19.5-19.5-45 19.5-45 45-19.5zM288 319.5q25.5 0 45 19.5t19.5 45-19.5 45-45 19.5-45-19.5-19.5-45 19.5-45 45-19.5zM352.5 576q0 25.5-19.5 45t-45 19.5-45-19.5-19.5-45 19.5-45 45-19.5 45 19.5 19.5 45z" />
+				</svg>
+				<span>{{ option.label }}</span>
+			</div>
+
+			<div class="rank-buttons">
+				<button @click="moveUp(index)" @mousedown="setHighlight(index)">
+					<svg xmlns="http://www.w3.org/2000/svg" width="15" height="15" viewBox="0 0 768 768">
+						<path d="M384 256.5l192 192-45 45-147-147-147 147-45-45z" />
+					</svg>
+				</button>
+
+				<button @click="moveDown(index)" @mousedown="setHighlight(index)">
+					<svg xmlns="http://www.w3.org/2000/svg" width="15" height="15" viewBox="0 0 768 768">
+						<path d="M531 274.5l45 45-192 192-192-192 45-45 147 147z" />
+					</svg>
+				</button>
+			</div>
+		</div>
+	</VueDraggable>
+
+	<ValidationMessage
+		:message="question.validationState.violation?.message.asString"
+		:show-message="touched || submitPressed"
+	/>
+</template>
+
+<style scoped lang="scss">
+@import 'primeflex/core/_variables.scss';
+
+$rankSpacing: 7px;
+$rankBorder: 1px solid var(--surface-200);
+$rankBorderRadius: 10px;
```

Comment:
Does `vue-draggable-plus` reference these variable names too? Or do they only affect the local styles? If they're just local, I'd prefer to use CSS custom properties ("variables") over Sass syntax. In general I'd like to try to keep Sass-specific syntax light, if we use it at all. CSS has come a long way, and Sass comes with a lot of surprises and a few headaches. In the long run, it would be nice to see if we can get away with eliminating (at least direct) Sass usage entirely.

Reply:
> Does vue-draggable-plus reference these variable names, too? 

No, it doesn't. These variables are for this specific component. 

> If they're just local, I'd prefer to use CSS custom properties ("variables") 

Sure! I have fixed it, and it is now using CSS variables. 
I have left the variable's definition in a separate block at the top. IMO, It's more organized that way. Let me know what you think.

Reply:
I think that's a great pattern! I've often ended up with something similar. Maybe even feels worth seeing if we can find a lint rule that helps us make it a convention.